syntax = "proto3";

package quick_verify_advanced;

enum RequestType {
    SendModel = 0;
    ReceiveModel = 1;
    ReceiveModelComplete = 2;
    Execute = 3;
    ExecuteComplete = 4;
 
    SendAndExecute = 5;

    Start = 6;
    TestExecute = 7;
    TestTransfer = 8;
    TestClear = 9;
}

message Request {
    RequestType type = 1;
    string payload = 2;

    SendModelProto send_model =3;
    ReceiveModelProto receive_model = 4;
    ReceiveModelCompleteProto receive_model_complete = 5;
    ExecuteProto execute = 6;
    ExecuteCompleteProto execute_complete = 7;
    SendAndExecuteProto send_and_execute = 8;
}

message SendModelProto{
    int32 transfer_block_id = 1;
    int32 node_id = 2;
}

message ReceiveModelProto{
    bytes param = 1;
    int32 transfer_block_id = 2;
    int32 node_id = 3;
}

message ReceiveModelCompleteProto{
    int32 transfer_block_id = 1;
    int32 src_node_id = 3;
    int32 dst_node_id = 4;
}



message SendAndExecuteProto{
    int32 cal_block_id = 1;
    int32 node_id = 2;
    int32 execute_id = 3;
}
message ExecuteProto{
    int32 cal_block_id = 1;
    bytes input_data = 2;
    int32 execute_id = 3;
}
message ExecuteCompleteProto{
    int32 cal_block_id = 1;
    bytes output_data = 2;
    bool is_large = 3;
    int32 execute_id = 4;
    int32 node_id = 5;
}